server/protocols/jmap/src/main/java/org/apache/james/jmap/model/mailbox/Rights.java

```diff
@@ -213,6 +217,36 @@ public MailboxACL toMailboxAcl() {
             .reduce(MailboxACL.EMPTY, union);
     }
 
+    public Optional<Boolean> mayReadItems(Username username) {
+        return containsRight(username, Right.Read);
+    }
+
+    public Optional<Boolean> mayAddItems(Username username) {
+        return containsRight(username, Right.Insert);
+    }
+
+    public Optional<Boolean> mayCreateChild(Username username) {
+        return UNSUPPORTED;
+    }
+
+    public Optional<Boolean> mayRemoveItems(Username username) {
+        return containsRight(username, Right.DeleteMessages);
+    }
+
+    public Optional<Boolean> mayRename(Username username) {
+        return containsRight(username, Right.Write);
+    }
+
+    public Optional<Boolean> mayDelete(Username username) {
+        return UNSUPPORTED;
+    }
+
+    private Optional<Boolean> containsRight(Username username, Right right) {
+        return Optional.ofNullable(rights.get(username))
+            .filter(rightList -> !rightList.isEmpty())
```

Comment:
I don't understand why this filter is useful

Reply:
Because if there is not entry, then the right list is an empty list.

This is an issue because the owner, supposed to have full rights, don't have specific ACL, so you need to resolve this at a higher level. Hence you filter them.